api/domain/user.go

```diff
@@ -11,6 +13,17 @@ type User struct {
 	Address       Address       `gorm:"embedded" json:"address"`
 	PaymentMethod PaymentMethod `gorm:"embedded" json:"payment_method"`
 	Photo         *string       `json:"photo,omitempty" example:"data:image/png;base64,mfkirjIDSFIj324if..."`
+	CreatedAt     time.Time     `json:"created_at" gorm:"not null" example:"2021-01-01T00:00:00Z"`
+}
+
+type UserWithReview struct {
+	FirstName           *string   `json:"first_name,omitempty" example:"John"`
+	LastName            *string   `json:"last_name,omitempty" example:"Doe"`
+	Photo               *string   `json:"photo,omitempty" example:"data:image/png;base64,mfkirjIDSFIj324if..."`
+	Rating              *float64  `json:"rating,omitempty" example:"4.5"`
+	NumberOfReviews     *int      `json:"number_of_reviews,omitempty" example:"52"`
+	NumberOfArtToysSold *int      `json:"number_of_art_toys_sold,omitempty" example:"52"`
+	CreatedAt           time.Time `json:"created_at" gorm:"not null" example:"2021-01-01T00:00:00Z"`
```

Comment:
ฝากเช็คว่า embedded ได้มั้ย
```suggestion
type UserWithReview struct {
        User
	NumberOfReviews     *int      `json:"number_of_reviews,omitempty" example:"52"`
	NumberOfArtToysSold *int      `json:"number_of_art_toys_sold,omitempty" example:"52"`
```

ละก็ merge ขอเจมเข้าด้วย

Reply:
embed ได้ แต่มันจะกลายเป็น user แล้วค่อยเข้าไป name photo อีกที

`{
            "User": {
                "id": 3849994009,
                "email": "jomlovejen7956@gmail.com",
                "first_name": "kamanun",
                "last_name": "maneesri",
                "address": {
                    "district": "bangkok"
                },
                "payment_method": {
                    "cvv": "123"
                },
                "photo": "jom.png",
                "created_at": "0001-01-01T00:00:00Z"
            },
            "rating": 0,
            "number_of_reviews": 0,
            "number_of_art_toys_sold": 0,
            "created_at": "2025-03-23T23:50:02.1640999+07:00"
        },
        {
            "User": {
                "id": 4285067867,
                "email": "kamanun.m@gmail.com",
                "address": {},
                "payment_method": {},
                "created_at": "0001-01-01T00:00:00Z"
            },
            "rating": 4.5,
            "number_of_reviews": 2,
            "number_of_art_toys_sold": 2,
            "created_at": "2025-03-23T23:49:39.3709886+07:00"
        }`